lib/mix/lib/mix/tasks/test.ex

```diff
@@ -293,9 +296,35 @@ defmodule Mix.Tasks.Test do
   It must return either `nil` or an anonymous function of zero arity that will
   be run after the test suite is done.
 
-  ## "Stale"
+  ## OS Processes Partitioning
+
+  While ExUnit supports the ability to run tests concurrently within the same
+  Elixir instance, it is not always possible to run all tests concurrently. For
+  example, some tests may rely on global resources.
+
+  For this reason, `mix test` all supports partitioning the test files across
+  different Elixir instances. This is done by setting the `--partitions` option
+  to an integer, with the number of partitions, and setting the `MIX_TEST_PARTITION`
+  environment variable to control which test partition that particular instance
+  is running. This can be also be useful if you want to distribute testing across
+  multiple machines.
+
+  For example, to split a test suite into 4 partitions and run them, you would
+  use the following commands:
 
-  The `--stale` command line option attempts to run only those test files which
+      MIX_TEST_PARTITION=1 mix test --partitions 4
```

Comment:
Explaining why this option is passed via an environment variable instead of a command-line option made me wonder if that's the right choice. 

I think this is the first time Mix uses environment variables to pass options that are particular to one Mix task; and users could define their own environment variable and use that as well in the configuration files and scripts:

```
mix test --num-partitions 4 --partition $MY_PARTITION
```

And have something in the documentation like:

> If the number of partitions or the partition must be accessed in configuration files and scripts, you can use environment variables:
>
>     mix test --partitions $MIX_TEST_NUM_PARTITIONS --partition $MIX_TEST_PARTITION
>

Said that, I don't have a strong opinion about this, but I thought the discussion was worthy 🙂

Reply:
Yeah, I thought about this approach, but I think it would be beneficial to standardize. For example, if we standardize, Phoenix generators can generate a `config/test.exs` like this:

    database: "demo_test#{System.get_env("MIX_TEST_PARTITION")}"

Without it, Phoenix would have to pick a name which may be inconsistent with the name other tools may pick.